packages/core/rxjs-interop/src/to_observable.ts

```diff
@@ -38,20 +38,23 @@ export function toObservable<T>(
     ): Observable<T> {
   !options?.injector && assertInInjectionContext(toObservable);
   const injector = options?.injector ?? inject(Injector);
+  const subject = new ReplaySubject<T>(1);
 
-  // Creating a new `Observable` allows the creation of the effect to be lazy. This allows for all
-  // references to `source` to be dropped if the `Observable` is fully unsubscribed and thrown away.
-  return new Observable(observer => {
-    const watcher = effect(() => {
-      let value: T;
-      try {
-        value = source();
-      } catch (err) {
-        observer.error(err);
-        return;
-      }
-      observer.next(value);
-    }, {injector, manualCleanup: true, allowSignalWrites: true});
-    return () => watcher.destroy();
+  const watcher = effect(() => {
+    let value: T;
+    try {
+      value = source();
+    } catch (err) {
+      untracked(() => subject.error(err));
+      return;
+    }
+    untracked(() => subject.next(value));
+  }, {injector, manualCleanup: true});
```

Comment:
Any reason why you choose to not write it this way ? 

```suggestion
    try {
      const value = source();
      untracked(() => subject.next(value));
    } catch (err) {
      untracked(() => subject.error(err));
    }
  }, {injector, manualCleanup: true});
```

or event `untracked(() => subject.next(source()));`

Reply:
Yes, see https://github.com/angular/angular/discussions/49681#discussioncomment-5514584.

1. We need to catch errors outside the `next` call so the `source()` read is outside `subject.next`
2. We don't want to catch errors that arise in the downstream observers so `subject.next` is outside the `try...catch`